--- gdb-8.2.orig/gdb/remote.c	2018-09-05 16:44:28.000000000 +0900
+++ gdb-8.2/gdb/remote.c	2018-11-05 15:30:39.745920969 +0900
@@ -8034,9 +8034,21 @@
   buf_len = strlen (rs->buf);
 
   /* Further sanity checks, with knowledge of the architecture.  */
-  if (buf_len > 2 * rsa->sizeof_g_packet)
-    error (_("Remote 'g' packet reply is too long (expected %ld bytes, got %d "
-	     "bytes): %s"), rsa->sizeof_g_packet, buf_len / 2, rs->buf);
+  if (buf_len > 2 * rsa->sizeof_g_packet) {
+  
+    rsa->sizeof_g_packet = buf_len;
+    for(i = 0; i < gdbarch_num_regs(gdbarch); i++){
+
+      if (rsa->regs[i].pnum == -1)
+	continue;
+
+      if (rsa->regs[i].offset >= rsa->sizeof_g_packet)
+	rsa->regs[i].in_g_packet = 0;
+      else
+	rsa->regs[i].in_g_packet = 1;
+    }
+  }
+
 
   /* Save the size of the packet sent to us by the target.  It is used
      as a heuristic when determining the max size of packets that the
